plonky3/src/verifier.rs

```diff
@@ -10,250 +12,423 @@ use p3_matrix::dense::RowMajorMatrixView;
 use p3_matrix::stack::VerticalPair;
 use tracing::instrument;
 
+use crate::circuit_builder::{PowdrCircuit, PowdrTable};
+use crate::params::{Challenge, Challenger, Commitment, Plonky3Field, ProverData};
 use crate::symbolic_builder::{get_log_quotient_degree, SymbolicAirBuilder};
-use crate::{MultiStageAir, Proof, StarkVerifyingKey, VerifierConstraintFolder};
+use crate::{
+    FieldElementMap, MultiStageAir, Proof, StageOpenedValues, StarkVerifyingKey, TableOpenedValues,
+    VerifierConstraintFolder,
+};
 use p3_uni_stark::{PcsError, StarkGenericConfig, Val};
 
-#[instrument(skip_all)]
-pub fn verify<SC, A>(
-    config: &SC,
-    air: &A,
-    challenger: &mut SC::Challenger,
-    proof: &Proof<SC>,
-    public_values: &Vec<Val<SC>>,
-) -> Result<(), VerificationError<PcsError<SC>>>
+/// A sub-table to be proven, in the form of an air and values for the public inputs
+struct Table<'a, T: FieldElementMap>
```

Comment:
why are this Table and prover::Table different?

Reply:
After this comment I realised they are too similar. I ended up refactoring by packing in `Table` everything which relates to this particular table within the verifier: the preprocessed commitments from the vk, the air, the opened values, the publics.